pennylane/interfaces/execution.py

```diff
@@ -589,12 +593,14 @@ def cost_fn(params, x):
            [ 0.        ,  0.        , -0.95533649]])
     """
     if qml.active_return():
+        grad_on_execution = mode == "forward" if mode != "best" else "best"

```

Comment:
Why is the `mode` keyword argument still in `execute`? If the intention is to avoid using ML language, then I think it might be even more important to change `mode` to `grad_on_execution` in `execute` as `_execute_new` is a private function and `execute` is the user-facing function.

Reply:
Because I don't want to go through a deprecation cycle for that, the function `_execute_new` will replace the current function in Q2 as the return type work is done.